addons/point_of_sale/models/pos_session.py

```diff
@@ -467,6 +497,24 @@ def _cannot_close_session(self):
             return {'successful': False, 'message': _("You cannot close the POS when orders are still in draft"), 'redirect': False}
         if self.state == 'closed':
             return {'successful': False, 'message': _("This session is already closed."), 'redirect': True}
+        if bank_payment_method_diffs:
+            no_loss_account = self.env['account.journal']
+            no_profit_account = self.env['account.journal']
```

Comment:
here you can do browse all payment methods and get all journal like this:

self.env['pos.payment.method'].browse(list(bank_payment_method_diffs.keys())).mapped('journal_id')

then you loop on the journals

Reply:
Doing that, I will loose the `payment_method_id`. I need it to get the diff_amount in the `compare_to_zero` line. 

```
compare_to_zero = self.currency_id.compare_amounts(bank_payment_method_diffs.get(payment_method_id), 0)
```

I sent again a direct message to Alain. Let's wait for a few while and see if he responds.

Reply:
But yeah, I'm calling browse in each loop. I'll change it.

Reply:
ok, right. Just do the browse before loop, and then loop on payment method